pandas/core/arrays/base.py

```diff
@@ -931,7 +931,7 @@ def copy(self) -> ABCExtensionArray:
         """
         raise AbstractMethodError(self)
 
-    def view(self, dtype=None) -> Union[ABCExtensionArray, np.ndarray]:
+    def view(self, dtype=None) -> ArrayLike:
```

Comment:
according to the docstring the return type should be just ExtensionArray?

Reply:
can you update either the doc-string or the type

Reply:
updated

Reply:
Can you indicate *when* it returns an ndarray? 

(I am wondering if we actually should be strict here, and tighten this method to always return an EA)

Reply:
lgtm outside of this comment

Reply:
> Can you indicate when it returns an ndarray?

It isn't obvious to me how; did you have something in mind?

> (I am wondering if we actually should be strict here, and tighten this method to always return an EA)

xref #24877 same idea but for astype.